src/main/java/com/thirdcc/webapp/service/impl/EventAttendeeServiceImpl.java

```diff
@@ -42,9 +51,23 @@ public EventAttendeeServiceImpl(EventAttendeeRepository eventAttendeeRepository,
     @Override
     public EventAttendeeDTO save(EventAttendeeDTO eventAttendeeDTO) {
         log.debug("Request to save EventAttendee : {}", eventAttendeeDTO);
-        EventAttendee eventAttendee = eventAttendeeMapper.toEntity(eventAttendeeDTO);
-        eventAttendee = eventAttendeeRepository.save(eventAttendee);
-        return eventAttendeeMapper.toDto(eventAttendee);
+
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.OPEN);
+            add(EventStatus.POSTPONED);
+        }};
+
+        Event event = eventRepository
+            .findOneByIdAndStatusIn(eventAttendeeDTO.getEventId(), eventStatuses)
+            .get();
```

Comment:
try to use 
`orElseThrow(() -> new BadRequestException())`
so no need to have a if else below to do checking

Reply:
to learn more about Optional API can refer to here. https://www.baeldung.com/java-optional

```
Event event = eventRepository
            .findOneByIdAndStatusIn(eventAttendeeDTO.getEventId(), eventStatuses)
            .get();
```
this code above is bad because it might throw `NullPointerException`